packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/Metric.tsx

```diff
@@ -0,0 +1,289 @@
+import type { CreateAlertDefinitionForm, MetricCriteriaForm } from '../types';
+import type {
+  AvailableMetrics,
+  MetricAggregationType,
+  MetricOperatorType,
+  MetricUnitType,
+} from '@linode/api-v4';
+import type { FieldPathByValue } from 'react-hook-form';
+
+interface MetricCriteriaProps {
+  /**
+   * apiErrors while accessing the metric definitions endpoint
+   */
+  apiError: boolean[];
+  /**
+   * metric data fetched by the metric definitions endpoint
+   */
+  data: AvailableMetrics[];
+  /**
+   * name (with the index) used for the component to set in form
+   */
+  name: FieldPathByValue<CreateAlertDefinitionForm, MetricCriteriaForm>;
+  /**
+   * function to delete the Metric component
+   * @returns void
+   */
+  onMetricDelete: () => void;
+  /**
+   * to control when to show the delete icon
+   */
+  showDeleteIcon: boolean;
+}
+export const Metric = (props: MetricCriteriaProps) => {
+  const { apiError, data, name, onMetricDelete, showDeleteIcon } = props;
+  const [isMetricDefinitionError, isMetricDefinitionLoading] = apiError;
+  const { control, setValue } = useFormContext<CreateAlertDefinitionForm>();
+
+  const handleDataFieldChange = (
+    selected: { label: string; unit: MetricUnitType; value: string },
+    operation: string
+  ) => {
+    const fieldValue: MetricCriteriaForm = {
+      aggregation_type: null,
+      dimension_filters: [],
+      metric: null,
+      operator: null,
+      threshold: 0,
+    };
+    if (operation === 'selectOption') {
+      setValue(name, {
+        ...fieldValue,
+        metric: selected.value,
+      });
+    }
+    if (operation === 'clear') {
+      setValue(name, fieldValue);
+    }
+  };
+
+  const metricOptions = React.useMemo(() => {
+    return data
+      ? data.map((metric) => ({
+          label: metric.label,
+          value: metric.metric,
+        }))
+      : [];
+  }, [data]);
+
+  const metricWatcher = useWatch({ control, name: `${name}.metric` });
+
+  const selectedMetric = React.useMemo(() => {
+    return data && metricWatcher
+      ? data.find((metric) => metric.metric === metricWatcher)
+      : null;
+  }, [data, metricWatcher]);
+
+  const unit = selectedMetric?.unit ?? null;
+  const aggOptions = React.useMemo((): Item<
+    string,
+    MetricAggregationType
+  >[] => {
+    return selectedMetric && selectedMetric.available_aggregate_functions
+      ? MetricAggregationOptions.filter((option) =>
+          selectedMetric.available_aggregate_functions.includes(option.value)
+        )
+      : [];
+  }, [selectedMetric]);
+
+  const serviceWatcher = useWatch({ control, name: 'serviceType' });
+  return (
+    <Box
+      sx={(theme) => ({
+        backgroundColor:
+          theme.name === 'light' ? theme.color.grey5 : theme.color.grey9,
+        borderRadius: 1,
+        p: 2,
+      })}
+      data-testid={`${name}-id`}
+    >
+      <Stack>
+        <Box display={'flex'} justifyContent="space-between">
```

Comment:
Use this syntax whenever possible

```suggestion
        <Box display="flex" justifyContent="space-between">
```

Reply:
Is there any lint rule that can be added to check this ? Sometimes the IDE directly adds it and it misses our internal checks. I will find and replace the current instances of this.